### ISSUE TITLE
Unable to build for debug

### ISSUE THREAD

**Issue by B1tBandit** [Starfield-Reverse-Engineering/CommonLibSF#239]
Preset: build-debug-msvc-msvc

Release builds are working great so far, but when I try to set breakpoints with a debug build, SFSE throws an error stating my plugin didn't load correctly.

At first I had a hard time creating a debug build because of the static asserts and _ITERATOR_DEBUG_LEVEL
`static_assert(sizeof(std::string) == 0x20);  // If this fails, _ITERATOR_DEBUG_LEVEL is greater than 0`

I was finally able to make a debug build by either commenting out static asserts or forcing _ITERATOR_DEBUG_LEVEL to 0. Both options result in SFSE errors when breakpoints are set. I confirmed the process was attached and symbols loaded. 

In summary..
Debug build -> Attach to process -> No breakpoints -> Plugin loads and logs
Debug build -> Attach to process -> Breakpoints set -> Plugin fails to load

Are there steps that I'm missing?

**Comment by Qudix** [comment]
This is a known problem with using msvc stl types in class definitions, the solution would be to re-implement these types based on the stl version the game was built with without any debug machinery.

**Comment by B1tBandit** [comment]
So debugging is out of the question unless I re-implement std::string and std::vector so the below lines pass?
```c++
static_assert(sizeof(std::string) == 0x20);  // If this fails, _ITERATOR_DEBUG_LEVEL is greater than 0
static_assert(sizeof(std::vector<int>) == 0x18);
```

Closing the issue. It would be much appreciated if someone could provide further guidance or alternatives.
